src/fixit/rules/pyflakes/no_double_import.py

```diff
@@ -0,0 +1,210 @@
+        double_imports: List[str] = []
+        for import_alias in node.names:
+            name = import_alias.asname.name if import_alias.asname is not None else import_alias.name
+            name_repr = self._get_name_repr(name)
+            if name_repr in self.double_imports[node]:
+                double_imports.append(name_repr)
+            else:
+                names_to_keep.append(import_alias.with_changes(comma=cst.MaybeSentinel.DEFAULT))
+        if len(names_to_keep) == 0:
+            self.report(node, message=self._message(double_imports), replacement=cst.RemovalSentinel.REMOVE)
+        else:
+            new_node = node.with_changes(names=names_to_keep)
+            self.report(node, message=self._message(double_imports), replacement=new_node)
+
+    def visit_Import(self, node: cst.Import) -> None:
+        self.get_double_import(node)
+
+    def visit_ImportFrom(self, node: cst.ImportFrom) -> None:
+        self.get_double_import(node)
+
+    def leave_Import(self, node: cst.Import) -> None:
+        return self.delete_double_import(node)
+
+    def leave_ImportFrom(self, node: cst.ImportFrom) -> None:
+        return self.delete_double_import(node)
+
+    VALID = [
+        Valid(
+            """
+            import a
+
+            x = a.b([1, 3, 5, 6, 8])
+            """
+        ),
+        Valid(
+            """
+            def fn1():
+                import a
+                return a.b
+
+            def fn2():
+                import a
+                return a.c
+            """
+        ),
+        Valid(
+            # valid, not good
+            """
+            import a as f
+            import a
+            """
+        ),
+        Valid(
+            """
+            import a
+            from a import b
+            """
+        ),
+        Valid(
+            """
+            import a
+            import a.b.c.d
+            """
+        ),
+        Valid(
+            """
+            import a
+            from a import *
+            """
+        ),
+        # This is allowable to cover the case that we have a `from a import fn`
+        # statement in another file 
+        Valid(
+        """
+            import a
+
+            def fn():
+                import a
+                return a.b
+        """
+        )
+    ]
+
+    INVALID = [
+        Invalid(
+            """
+            import a
+            import b
+            import a
+            """,
+            expected_replacement="""
+            import a
+            import b
+            """,
+        ),
+        Invalid(
+            """
+            import g, e as g
+
+            x = g.b([1, 3, 5, 6, 8])
+            """,
+            expected_replacement="""
+            import e as g
+
+            x = g.b([1, 3, 5, 6, 8])
+            """,
+        ),
+        Invalid(
+            """
+            import a, b, a
+            """,
+            expected_replacement="""
+            import a, b
+            """,
```

Comment:
I know in this case, both `a` imports are referring to the same module, but I don't like the idea that we're removing the *second* import. We should instead be removing the first one, so that we're giving consistent runtime behavior before/after. For imports with side-effects, order unfortunately matters.